src/lib/containers/access_requirement_list/SelfSignAccessRequirement.tsx

```diff
@@ -1,12 +1,139 @@
 }
 
-export default function SelfSignAccessRequirement({
+export default function SelfSignAccessRequirementComponent({
   accessRequirement,
-}: SelfSignAccessRequirement) {
-  return <div></div>
+  token,
+  user,
+  onHide,
+}: Props) {
+  const [wikiPage, setWikiPage] = useState<WikiPageKey | undefined>(undefined)
+  const [userBundle, setUserBundle] = useState<UserBundle | undefined>(
+    undefined,
+  )
+  const [accessRequirementStatus, setAccessRequirementStatus] = useState<
+    AccessRequirementStatus | undefined
+  >(undefined)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const accessRequirementText = 'Self Sign Access Requirement'
+
+  useEffect(() => {
+    const getSelfSignAccessData = async () => {
+      try {
+        setIsLoading(true)
+        const wikiPageRequirment = await SynapseClient.getWikiPageKey(
+          token,
+          accessRequirement.id,
+        )
+
+        setWikiPage(wikiPageRequirment)
+
+        const certificationOrVerification =
+          SynapseConstants.USER_BUNDLE_MASK_IS_CERTIFIED |
+          SynapseConstants.USER_BUNDLE_MASK_IS_VERIFIED
+
+        if (user && user.ownerId) {
+          const bundle = await SynapseClient.getUserBundle(
+            user.ownerId,
+            certificationOrVerification,
+            token,
+          )
+          setUserBundle(bundle)
+        }
+
+        const selfSignAccessRequirement = await SynapseClient.getAccessRequirementStatus(
+          token,
+          accessRequirement.id,
+        )
+        setAccessRequirementStatus(selfSignAccessRequirement)
+      } catch (err) {
+        console.error('Error on get Self Sign Access Requirement: ', err)
+      } finally {
+        setIsLoading(false)
+      }
+    }
+
+    getSelfSignAccessData()
+  }, [accessRequirement, token, user])
+
+  return (
+    <div>
+      {accessRequirement.isCertifiedUserRequired && (
+        <div className="requirement-container">
+          <AccessApprovalCheckMark isCompleted={userBundle?.isCertified} />
+          <div>
+            <p className="self-sign-access-title bold-text">
+              You must first become a
+              <a
+                className="self-sign-access-certified bold-text"
+                href="https://www.synapse.org/#!Quiz:"
+              >
+                &nbsp;certified user
+              </a>
+            </p>
+            {isLoading && <span className="spinner" />}
+
+            <p
+              className={`self-sign-access-certified-success-text ${
+                userBundle?.isCertified ? 'show' : 'hide'
+              }`}
+            >
+              You have became a certified user
+            </p>
+          </div>
+        </div>
+      )}
+      {accessRequirement.isValidatedProfileRequired && (
+        <div className="requirement-container">
+          <AccessApprovalCheckMark isCompleted={userBundle?.isVerified} />
+          <div>
+            <p className="self-sign-access-title bold-text">
```

Comment:
this should only show if they haven't verified right? right now it will always show